cmd/gardener-controller-manager/app/options.go

```diff
@@ -68,6 +69,11 @@ func (o *options) complete() error {
 }
 
 func (o *options) validate() error {
+	// Add feature flags before validating the config file, as they might influence the validation logic.
+	if err := features.DefaultFeatureGate.SetFromMap(o.config.FeatureGates); err != nil {
+		return err
+	}
```

Comment:
Shouldn't we rather move this to the `app.go` before the `opts.complete()` is called? It seems weird/wrong to have it here in `validate()`, don't you agree?

Reply:
> Calling `SetFromMap` in `run` is too late. The config file validation will complain about the IPv6 `seedConfig` if the `IPv6SingleStack` feature gate is not enabled (via `SetFromMap`) beforehand.

ref https://github.com/gardener/gardener/pull/7561#discussion_r1155738146

Reply:
I am not suggesting to call it in `run()` but before `opts.complete()` which is called here: https://github.com/gardener/gardener/blob/7b30c1147001d7ceaab6b822ad171be26e4cd12f/cmd/gardener-controller-manager/app/app.go#L62

Reply:
Ah sorry, I misread your comment.
You're right., `SetFromMap` should rather be done in `complete()` than in `validate()`.
Calling it before `complete()` is too early, the config is only decoded in `complete()`.

I will move the `SetFromMap` calls to `complete()`.

Reply:
Done in a new commit, tell me if you like it :)